src/ts/components/settingspanel.ts

```diff
@@ -72,28 +103,132 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
       this.onHide.subscribe(() => {
         // Clear timeout when hidden from outside
         this.hideTimeout.clear();
+        // Reset navigation
+        this.resetNavigation();
       });
     }
 
-    // Fire event when the state of a settings-item has changed
-    let settingsStateChangedHandler = () => {
+    // pass event from root page through for backwards compatibility
+    this.getRootPage().onSettingsStateChanged.subscribe(() => {
       this.onSettingsStateChangedEvent();
+    });
 
-      // Attach marker class to last visible item
-      let lastShownItem = null;
-      for (let component of this.getItems()) {
-        component.getDomElement().removeClass(this.prefixCss(SettingsPanel.CLASS_LAST));
-        if (component.isShown()) {
-          lastShownItem = component;
-        }
-      }
-      if (lastShownItem) {
-        lastShownItem.getDomElement().addClass(this.prefixCss(SettingsPanel.CLASS_LAST));
+    this.updateActivePageClass();
+  }
+
+  private updateActivePageClass(): void {
+    this.getPages().forEach((page: SettingsPanelPage, index) => {
+      if (index === this.activePageIndex) {
+        page.getDomElement().addClass(SettingsPanel.ACTIVE_PAGE_CLASS);
```

Comment:
You need to prefix with `this.prefixCss`, also in any other place where this class is used.